miqa/core/rest/experiment.py

```diff
@@ -79,12 +80,18 @@ def create(self, request, *args, **kwargs):
         project = Project.objects.get(id=serializer.data['project'])
         if not get_perms(request.user, project):
             Response(status=status.HTTP_401_UNAUTHORIZED)
-        experiment = Experiment(
-            project=project,
-            name=serializer.data['name'],
-            lock_owner=None,
-        )
-        experiment.save()
+        try:
+            experiment = Experiment(
+                project=project,
+                name=serializer.data['name'],
+                lock_owner=None,
+            )
+            experiment.save()
+        except utils.IntegrityError:
+            raise APIException('An experiment with the same name already exists in this project.')
+        except Exception as e:
+            print(type(e))
```

Comment:
Should this block go away? Seems like we shouldn't be hiding unexpected exceptions here.

Reply:
We still raise an APIException with the exact text from the original exception. This is the catch-all that will raise any other types of errors to the user interface. The IntegrityError block just applies a custom error text instead of the default text about constraint violation. Although, I hadn't meant to leave the print statement in there...

Reply:
I think that's not a good practice. If these errors are truly unexpected, we don't want to be sending the exception data to the end-user, because it could contain things we don't want to leak out. (Not to mention it's likely overly technical, and our developers who can make sense of that will be getting the technical details anyway from Sentry.)

I think we should do our best to try and capture any *known* invalid behaviors and wrap them up with a nice error message, but for truly unexpected exceptions, we need to just log them in Sentry and send a generic message to the user IMO.

Reply:
Okay, that's a good explanation. For now, we may have to tell Killian the same thing, that whatever exception he was getting is still unknown and he may need to tell us how to replicate the issue (if the problem was truly different than the experiment name constraint)

Reply:
Could we output to the user, and log, a reference id (just a random number/string)? That way, Killian can tell us the reference number and we (with Jim's help) can look up the details of the error in the logs.

Reply:
I like this idea and have done this on other projects by installing an exception handler generally across the API that handles otherwise uncaught `Exception`s. It creates a `uuid4` and sends it with the error message, and then calls `logger.exception()` including that UUID.

Reply:
Added in [11dd78c](https://github.com/OpenImaging/miqa/pull/412/commits/11dd78c7383708f92054ed5c4b1cf9b60288da83)